File: android/app/src/main/java/com/billsafe/data/entities/Subscription.kt
package com.billsafe.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity(tableName = "subscriptions")
data class Subscription(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val userId: String,
    val appName: String,
    val amount: Double,
    val billingCycle: String, // monthly, yearly, etc.
    val startDate: LocalDateTime,
    val renewalDate: LocalDateTime,
    val status: String = "active", // active, cancelled, expired
    val isUsed: Boolean = true,
    val lastUsedDate: LocalDateTime? = null,
    val paymentMethod: String = "",
    val createdAt: LocalDateTime = LocalDateTime.now()
)

File: backend/src/models/Subscription.js
const mongoose = require('mongoose');

const subscriptionSchema = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    appName: {
        type: String,
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    billingCycle: {
        type: String,
        enum: ['monthly', 'quarterly', 'yearly', 'one-time'],
        default: 'monthly'
    },
    startDate: {
        type: Date,
        required: true
    },
    renewalDate: {
        type: Date,
        required: true
    },
    status: {
        type: String,
        enum: ['active', 'cancelled', 'expired'],
        default: 'active'
    },
    isUsed: {
        type: Boolean,
        default: true
    },
    lastUsedDate: Date,
    paymentMethod: String,
    createdAt: {
        type: Date,
        default: Date.now
    }
});

// Index for efficient queries
subscriptionSchema.index({ userId: 1, status: 1 });
subscriptionSchema.index({ userId: 1, renewalDate: 1 });

module.exports = mongoose.model('Subscription', subscriptionSchema);
